### ISSUE TITLE
IEDriverServer doesn't use the SocksProxy settings

### ISSUE THREAD

**Issue by tarlabs** [SeleniumHQ/selenium#735]
Steps to reproduce:
Run the below C# code
```
		Proxy GetProxy(string IP, string port) {
			return new Proxy{
				Kind = ProxyKind.Manual,
				IsAutoDetect = false,
				SocksProxy = IP + ":" + port
			};
		}
				
		void BtnLaunchIEClick(object sender, EventArgs e)
		{
			var proxy = GetProxy("127.0.0.1", "8080");
			
			var options = new InternetExplorerOptions();
			
			options.Proxy = proxy;
			options.UsePerProcessProxy = false;
			
			var driver  = new InternetExplorerDriver(options);
			driver.Url = "http://www.tarlabs.com";
		}
```

Expected: 
Socks Proxy will be set

Actual:
Socks Proxy settings is not set

Issue:

```
std::wstring ProxyManager::BuildProxySettingsString() {
  LOG(TRACE) << "ProxyManager::BuildProxySettingsString";
  std::string proxy_string = "";
  if (this->proxy_type_ == WD_PROXY_TYPE_MANUAL) {
    if (this->http_proxy_.size() > 0) {
      proxy_string.append("http=").append(this->http_proxy_);
    }
    if (this->ftp_proxy_.size() > 0) {
      if (proxy_string.size() > 0) {
        proxy_string.append(" ");
      }
      proxy_string.append("ftp=").append(this->ftp_proxy_);
    }
    if (this->ssl_proxy_.size() > 0) {
      if (proxy_string.size() > 0) {
        proxy_string.append(" ");
      }
      proxy_string.append("https=").append(this->ssl_proxy_);
    }
  } else if (this->proxy_type_ == WD_PROXY_TYPE_AUTOCONFIGURE) {
    proxy_string = this->proxy_autoconfigure_url_;
  } else {
    proxy_string = this->proxy_type_;
  }
  LOG(DEBUG) << "Built proxy settings string: '" << proxy_string << "'";
  return StringUtilities::ToWString(proxy_string);
}
```
There is no check for the Socks Proxy settings. It would need below kind of code
```
proxy_string.append("socks=").append(this->socks_proxy_);
```

@jimevans, I don't have Visual Studio set in my environment, so I can't set fix it up and make a pull request. Can you check this out?

Regards,
Tarun

**Comment by jimevans** [comment]
It's not quite that simple, because it's complicated by the fact that WebDriver supports SOCKS proxy authentication. Unfortunately, while the IE driver will set proxy settings on a per-session basis, there are no Windows APIs that allow us to do that with the proxy authentication. Using SOCKS proxy authentication in conjunction with the `ie.usePerProcessProxy` capability is going to be messy. I'll put in a change to check for the SOCKS proxy configuration settings, but anyone using the combination of SOCKS proxy authentication and the aforementioned capability is going to be in largely unsupported territory.

Additionally, note that I have absolutely no way to validate that the SOCKS proxy settings are properly used by the browser, as I don't have a setup for a SOCKS proxy, nor am I willing to expend the effort to test this that thoroughly. If, for some reason, this doesn't work for you, you'll need to obtain a Visual Studio version, and attempt to make the changes yourself. I recommend Visual Studio 2013 Community Edition (not "Express"), which is completely free, and should give you the ability to build the IE driver from scratch.

I'll update this issue report when I'm done with the modifications.

**Comment by tarlabs** [comment]
@jimevans, I see very less cases of Password based Socks Proxy. So my suggestion would be that if someone sets the SocksUsername/SocksPassword then WebDriver can throw an error saying that it is not supported. Also about testing it with a sock proxy, it is quite easy if you have access to any VPS box. You will need to launch a dynamic port forwarding like below

```
ssh -D 8080 username@hostname cat -
```
This will create a local socks proxy on your machine

**Comment by jimevans** [comment]
Merged in 931f51a
